Introduction to xemc3.ipynb

```diff
@@ -0,0 +1,3171 @@
+       "     long_name:  Electron temperature\n",
+       "     units:      eV)"
+      ]
+     },
+     "execution_count": 15,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "R = ds.R_bounds.mean(dim=(\"delta_r\", \"delta_theta\", \"delta_phi\"))\n",
+    "z = ds.z_bounds.mean(dim=(\"delta_r\", \"delta_theta\", \"delta_phi\"))\n",
+    "phi = ds.phi_bounds.mean(dim=\"delta_phi\")\n",
+    "x = R * np.cos(phi)\n",
+    "y = R * np.sin(phi)\n",
+    "x, ds.Te"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 125,
+   "id": "b2b57e7d",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Total number of gridpoints in the W7X gridstructure is  2562048\n"
+     ]
+    }
+   ],
+   "source": [
+    "x.shape\n",
+    "tot_grid_points = x.shape[0]*x.shape[1]*x.shape[2]\n",
+    "print(\"Total number of gridpoints in the W7X gridstructure is \", tot_grid_points)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "13ee82c1",
+   "metadata": {},
+   "source": [
+    "## Use of NaN values in the parameter mesh\n",
+    "\n",
+    "Not all gridcells have a defined parameter value attached to it. This is mostly the outer and inner region of the machine where the values of many parameter has been left out because this is not the regions where the interesting physics happen.\n",
+    "This is illustrated in the above plot example of the electron temperature $T_e$. In the cell below you can se how large a fraction of the total number of gridpoints the parameter mesh for the electron temperature that has NaN as a value."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 130,
+   "id": "0b63deae",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "How many nans in the Te mesh?  (584972,)\n",
+      "Fraction of nans with respect to gridpoints  0.22832202987609912\n"
+     ]
+    }
+   ],
+   "source": [
+    "n_nans_Te = int(list(np.where(np.isnan(np.asarray(ds.Te)) == True)[0].shape)[0])\n",
+    "print(\"How many nans in the Te mesh? \", np.where(np.isnan(np.asarray(ds.Te)) == True)[0].shape)\n",
+    "print(\"Fraction of nans with respect to gridpoints \", n_nans_Te/tot_grid_points)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 104,
+   "id": "5e68a556",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "(139, 512, 36)"
+      ]
+     },
+     "execution_count": 104,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "np.asarray(ds.Te).shape"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9fcdc958",
+   "metadata": {},
+   "source": [
+    "## Grid\n",
+    "\n",
+    "In the cell below there is an interactive plot of the grid. You can use the slider to iterate through all toroidal slices(all $\\phi$ angles)."
```

Comment:
Actually, you are missing the last value.
All values [0, np.pi/5] are supported.